docs/io.cozy.files.md

```diff
@@ -27,14 +27,29 @@ The attributes of a file are:
 - `name`: {string} the file name
 - `trashed`: {bool} true if the file is in the trash
 - `md5sum`: {string} the checksum of its content, computed with the MD5 algorithm
-- `created_at`: {timestamp} the date of the creation of this directory
-- `updated_at`: {timestamp} the date of the last update of this directory
+- `created_at`: {date} the date of the creation of this directory
+- `updated_at`: {date} the date of the last update of this directory
 - `tags`: {array of strings} a list of tags
 - `size`: {number} the size of its content, in bytes
 - `executable`: {bool} true is the file has the executable bit on UNIX (`chmod +x`)
 - `class`: {string} a class in the list: `['image', 'document', 'audio', 'video', 'text', 'binary']`
 - `mime`: {string} the full mime-type
 - `metadata`: {map} an optional map of metadata (for example `width`, `height`, and `datetime` for an image)
+  - `width`: {number}
+  - `height`: {number}
+  - `datetime`: {date} : date in original image file metadata
+  - `gps`: {map} : localization metadata with the following attributes
+    - `lat`: {float}: latitude
+    - `long`: {float}: longitude
+    - `city`: {string}: nearest city (optional)
+    - `zip`: {string}: postal code of the nearest city (optional)
+    - `country`: {string}: name of the associated country if any (optional)
+  - `persons`: {array}: the maps can have the following attributes (optional)
```

Comment:
We don't  have `persons` currently, and only the stack can write the metadata of a file. What do you plan for this field? Is it possible to use another doctype for the `persons` on a photo? In particular, when a photo is shared (Cozy to Cozy sharings), does this people list should be sent to the recipients of the sharing.

Reply:
I actually did not know we needed to update the stack for this.
The doctype for the persons on the photo could be io.cozy.contacts, eventually.

And the question about the sharing is also a good one. Shouldn't we ask the use what he wants here? It also applies to localization metadata, no?

Since this causes more questions than what I expected, I propose to just handle localization metadata here I will create another PR regarding persons.

Reply:
For the localization, currently, it is extracted from the EXIF, and if the file is shared, its content is not modified and the EXIF will still be here. So, it's available to the recipients of the sharing.

Reply:
For me it's not a problem to share the `persons` metadata, as it does not disclose more information than the photo already gives: if my name had been found by facebook, my face is visible enough to be recognized by the recipients.
@nono wouldn't it be possible to add a route to put new metadata to a file, or to force them at the file creation? Beyond the `persons` case it would also be useful for the `gps` info, as facebook wipes the EXIF, but restitutes them through their API.

Reply:
It's possible. You can ask @poupotte to add that in our backlog

Reply:
https://github.com/cozy/cozy-stack/pull/1736